### PR TITLE
Protect against invalid readyweapon

### LINKED ISSUE #1245: MBF21: SSG autoswitch causes weapons to disappear
If the last two shells are fired immediately after automatically switching to SSG, the game will crash with an access violation. Similar to DSDA bug reported [here](https://github.com/kraflab/dsda-doom/issues/431), and the demos recorded with DSDA will cause a crash in Woof.

Unlike DSDA, in Woof the crash can be consistently reproduced without the need for tricky timing by using a custom dehacked SSG. To force a crash, the last two shells need to be fired separately (i.e. fired rather than refired).
Also unlike DSDA, on playback, the demo recorded in Woof simply ends, without causing a crash (briefly shows "game saved (suppressed)" at the end).

Can't seem to replicate in versions prior to 11.0. Additionally, in 12.0 there seems to be some issue with autoswitching from the rocket launcher as well (stays up with 0 ammo, see demo).

[Demos + custom SSG.](https://github.com/fabiangreffrath/woof/files/13205086/mbf21_ssg_woof.zip)

### PR DESCRIPTION
**Pull request by fabiangreffrath**
Fixes #1245

### CODE REVIEW

**Review by rfomin — APPROVED** [review]

**Comment by fabiangreffrath** [comment]
We'll wait for @kraflab's approval here as well.

**Review by kraflab — APPROVED** [review]

**Comment by fabiangreffrath** [comment]
Thank you @kraflab. Is there anything else from the DSDA 0.28 release cycle that we might want to apply to Woof?

**Comment by kraflab** [comment]
I think you've migrated the important stuff, I'll take a look and let you know if there's any other big stuff.

**Pull request merged**
